webauthn/webauthn.py

```diff
@@ -1017,7 +1043,10 @@ def verify(self):
             #             or not, is Relying Party-specific.
             sc = decoded_a_data[33:37]
             sign_count = struct.unpack('!I', sc)[0]
-            if sign_count or self.webauthn_user.sign_count:
+            if sign_count:
```

Comment:
How about we change this to `if not sign_count or not self.webauthn_user.sign_count:`? Then, you can raise your `WebAuthnUserDataMissing` exception and we can assume the `sign_count` is present further down.

Reply:
With something like 
```
            sc = decoded_a_data[33:37]
            sign_count = struct.unpack('!I', sc)[0]
            if not sign_count or not self.webauthn_user.sign_count:
                raise WebAuthnUserDataMissing("sign_count missing")

            if sign_count <= self.webauthn_user.sign_count:
                raise AuthenticationRejectedException(
                    'Duplicate authentication detected.')
```
We will have a `WebAuthnUserDataMissing` if the key does not provide a `sign_count` correctly, and the error would be a `WebAuthnUserDataMissing` which could let think that the error would come from the `WebAuthnUser` object (server side).

What do you think ?

Reply:
Good point, how about something like the following?

```
            sc = decoded_a_data[33:37]
            sign_count = struct.unpack('!I', sc)[0]

            if not sign_count:
                raise AuthenticationRejectedException("Unable to parse sign_count.")

            if not self.webauthn_user.sign_count:
                raise WebAuthnUserDataMissing("sign_count missing from WebAuthnUser.")

            if sign_count <= self.webauthn_user.sign_count:
                raise AuthenticationRejectedException(
                    'Duplicate authentication detected.')
```